R/rowWilcoxonTests.R

```diff
@@ -72,26 +71,152 @@
 #' cls_perm <- replicate(B, sample(cls))
 #' system.time(fwt <- rowWilcoxonTests(mat, categ = cls_perm, alternative = "two.sided"))
 #' 
-rowWilcoxonTests <- function(mat, categ, 
-                              alternative = c("two.sided", "less", "greater"), 
-                              correct = TRUE) {
+
+rowWilcoxonTests <- function (mat, categ, alternative = c("two.sided", "less", "greater"), 
+                                correct = TRUE) 
+{
+    alternative <- match.arg(alternative)
+    stopifnot(all(categ %in% c(0, 1)))
+    categ <- as.matrix(categ)
+    levels(categ) <- NULL
+    
+    rks <- rowRanks(mat, ties.method = "average")
+    ny <- sum(categ[,1] == 0) 
+    nx <- sum(categ[,1] == 1) 
+    min_stat <- nx * (nx + 1)/2 
+    
+    stats <- rks %*% categ
+    stats <- stats - min_stat
+    
+    rks2 <- rks 
+    mode(rks2) <- "integer"
+    n_ties <- rowTabulates(rks2) 
+    ties <- rowSums(n_ties^3 - n_ties) 
+    sigma <- sqrt((nx * ny/12) * ((ny + nx + 1) - ties/((ny + nx) * (ny + nx - 1)))) 
+    z <- stats - ny * nx/2
```

Comment:
Here one should be extra careful when `stats`  is a matrix and `nx` and `ny` are vectors!
I guess something like 

```
sweep(stats, MARGIN = 2, STATS = ny * nx / 2, FUN = "-")
```

should do the trick?